### PR TITLE
Implement fold for `KeccakProofInputs`

### PR DESCRIPTION
**Pull request by querolita**
This replicates what's done in `mips` for `ProofInputs`.

### CODE REVIEW

File: optimism/src/keccak/column.rs [hunk 1]
```diff
@@ -313,7 +338,7 @@ impl<G: Send + std::fmt::Debug> FromParallelIterator<G> for KeccakColumns<G> {
         I: IntoParallelIterator<Item = G>,
     {
         let mut iter_contents = par_iter.into_par_iter().collect::<Vec<_>>();
-        let next = iter_contents
+        let next: [G; ZKVM_KECCAK_COLS_NEXT] = iter_contents
```
Comment (dannywillems) [inline optimism/src/keccak/column.rs:341]: I would suppose the typing annotation is required, therefore you added it?

File: optimism/src/keccak/proof.rs [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
-    _evaluations: KeccakColumns<Vec<G::ScalarField>>,
+    evaluations: KeccakColumns<Vec<G::ScalarField>>,
 }
 
 impl<G: KimchiCurve> Default for KeccakProofInputs<G> {
     fn default() -> Self {
         KeccakProofInputs {
-            _evaluations: KeccakColumns {
+            evaluations: KeccakColumns {
                 hash_index: (0..1 << 15).map(|_| G::ScalarField::zero()).collect(),
```
Comment (dannywillems) [inline optimism/src/keccak/proof.rs:25]: Note: we should have a global variable, shared with proof in mips.

File: optimism/src/keccak/proof.rs [hunk 3]
```diff
@@ -49,3 +59,50 @@ pub struct KeccakProof<G: KimchiCurve, OpeningProof: OpenProof<G>> {
     _zeta_omega_evaluations: KeccakColumns<G::ScalarField>,
     _opening_proof: OpeningProof,
 }
+
+pub fn fold<
```
Comment (dannywillems) [inline optimism/src/keccak/proof.rs:63]: Can you comment succinctly what the function does please?

File: optimism/src/keccak/column.rs (hunk 1 above)
Comment (querolita) [reply optimism/src/keccak/column.rs:341]: Oh it was probably not needed and I just forgot it was there.

File: optimism/src/keccak/proof.rs (hunk 3 above)
Comment (querolita) [reply optimism/src/keccak/proof.rs:63]: At this point this is just aggregating instances together with a random combination. It does not contain folding equations per se, those will be located in the folding part of the project (still to be merged completely).

**Review by dannywillems — APPROVED** [review]

**Pull request merged**